src/App.tsx

```diff
@@ -1,61 +1,115 @@
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User, Todo } from './types';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUserById = (userId: number) => (
+  usersFromServer.find(user => user.id === userId) as User
+);
+
+const initTodos: Todo[] = todosFromServer.map(todo => ({
+  user: getUserById(todo.userId),
+  ...todo,
+}));
+
+export const App: React.FC<{}> = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUser, setUser] = useState('');
+  const [todos, setTodos] = useState(initTodos);
+  const [showTitleError, setTitleError] = useState(false);
+  const [showUserError, setUserError] = useState(false);
+
+  const generateId = () => (
+    Math.max(...todos.map(todo => todo.id)) + 1
+  );
+
+  const createTodo = () => {
+    setTitleError(!title);
+    setUserError(!selectedUser);
+
+    if (title && selectedUser) {
+      setTodos(state => (
+        state.concat({
+          id: generateId(),
+          user: getUserById(+selectedUser),
+          userId: +selectedUser,
+          completed: false,
+          title,
+        })
+      ));
+
+      setTitle('');
+      setUser('');
+    }
```

Comment:
wait, `.concat` mutates the state - we shouldn't do that
```suggestion
    if (title && selectedUser) {
      const newTodo = {
          id: generateId(),
          user: getUserById(+selectedUser),
          userId: +selectedUser,
          completed: false,
          title,
        }
      
      setTodos([
        ...todos,
        newTodo
      ]);

      setTitle('');
      setUser('');
    }
```